hs_app_timeseries/forms.py

```diff
@@ -1,32 +1,61 @@
+
 from hs_core.forms import BaseFormHelper
+from models import Site, Variable, Method, ProcessingLevel, TimeSeriesResult
+
 
 class SiteFormHelper(BaseFormHelper):
-    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, element_name=None,  *args, **kwargs):
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, element_name=None,
+                 *args, **kwargs):
 
-        # the order in which the model fields are listed for the FieldSet is the order these fields will be displayed
+        # the order in which the model fields are listed for the FieldSet is the order these
+        # fields will be displayed
         field_width = 'form-control input-sm'
         layout = Layout(
                         Field('site_code', css_class=field_width),
                         Field('site_name', css_class=field_width),
                         Field('organization', css_class=field_width),
                         Field('elevation_m', css_class=field_width),
-                        Field('elevation_datum', css_class=field_width),
-                        Field('site_type', css_class=field_width),
+                        Field('elevation_datum', css_class=field_width,
+                              title="Select 'Other...' to specify a new elevation datum term"),
+                        Field('site_type', css_class=field_width,
+                              title="Select 'Other...' to specify a new site type term"),
                  )
 
-        super(SiteFormHelper, self).__init__(allow_edit, res_short_id, element_id, element_name, layout,  *args, **kwargs)
+        super(SiteFormHelper, self).__init__(allow_edit, res_short_id, element_id, element_name,
+                                             layout,  *args, **kwargs)
 
 
 class SiteForm(ModelForm):
     def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, **kwargs):
+        self.cv_site_types = list(kwargs.pop('cv_site_types'))
+        self.cv_elevation_datums = list(kwargs.pop('cv_elevation_datums'))
         super(SiteForm, self).__init__(*args, **kwargs)
         self.helper = SiteFormHelper(allow_edit, res_short_id, element_id, element_name='Site')
 
+    def set_dropdown_widgets(self, site_type, elevation_datum):
+        cv_site_type_choices = _get_cv_dropdown_widget_items(self.cv_site_types, site_type)
+        self.fields['site_type'].widget = forms.Select(choices=cv_site_type_choices)
+
+        cv_e_datum_choices = _get_cv_dropdown_widget_items(self.cv_elevation_datums,
+                                                           elevation_datum)
+        self.fields['elevation_datum'].widget = forms.Select(choices=cv_e_datum_choices)
+
+    @property
+    def form_id(self):
+        form_id = 'id_site_%s' % self.number
+        return form_id
+
+    @property
+    def form_id_button(self):
+        form_id = 'id_site_%s' % self.number
+        return "'" + form_id + "'"
```

Comment:
This looks like it could be rendered more compactly as a single string interpolation.